### DISCUSSION TITLE
Calculate the MIllivolt Trimpot value from the inH2O

### DISCUSSION THREAD

**Discussion by KirikauKiwi** [DIY-Flow-Bench/DIY-Flow-Bench#254] [Code Development]
Can you check and verify this formula works for you to calculate the MV_TRIMPOT value for those using the MPXV7007 pressure sensors.

Millivolt offset = (inH2O/(0.65626025/249.08891) * 0.0001875), simply put in the inH2O that is currently displayed for your sensor to calculate the offset in millivolts, this value can then be entered for the MV_TRIMPOT currently found in the configuration.h for the relevant sensor.

- If your inH20 is displaying as a negative value your MV_TRIMPOT value will need to be entered as positive value.
- If your inH20 is displaying as a positive value your MV_TRIMPOT value will need to be entered as a negative value.

**Comment by DeeEmm** [comment]
The caveat with doing this is that it doesn't take into account the reference voltage, which is part of the calculation to convert sensor volts to kPa...

```
// Vout = Vcc x (0.057 x sensorVal + 0.5) --- Transfer function formula from MPXV7007DP Datasheet
sensorVal = ((sensorVolts / _hardware.get5vSupplyVolts()) -0.5) / 0.057;
```

Using this method, any deviation in reference voltage from nominal will cause an error. Of course this error may be insignificant, but it also may not be. Best practice is for the trimpot to trim only sensor voltage errors, not sensor voltage + reference voltage errors together.

Ideally the value should be set by measurement. However the easiest method is to report the voltage to the serial monitor as an API call and then the user can feed that back into the setup.

i.e. sensor voltage reported when reading 0kPa or for a given Baro reading

We can definitely look at this as part of the initial set up but need to consider that the target will be different for each sensor. For differential sensors it may be a mid range value, for regular sensors a lower voltage.

I've updated the API to include voltage values for pDiff and pRef. This will at least allow the user to report the static reading.

**Comment by DeeEmm** [comment]
Latest update includes API items for reporting pDiff and pRef voltage to serial monitor

24112402 - #213 Pitot / differential zeroing
                - #254 Report sensor voltage to API
                - Cookie to remember tile status
